src/screens/Survey/index.tsx

```diff
@@ -11,42 +18,73 @@ export const surveyScreenTestIds = {
   title: 'survey__title-text',
   description: 'survey__description-text',
   startSurveyButton: 'survey__start-survey-button',
+  loadingDialog: 'survey__loading-dialog',
+  toast: 'survey__toast',
 };
 
 const SurveyScreen = (): JSX.Element => {
+  const { id } = useParams();
+
+  const { survey, isLoading, isError } = useAppSelector((state) => state.survey);
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  function goBack() {
+    navigate(-1);
+  }
+
+  useEffect(() => {
+    dispatch(getSurveyAsyncThunk(id ?? ''));
+  }, [dispatch, id]);
+
+  useEffect(() => {
+    if (isError) {
+      toast.error('There is something wrong. Please try again later!', { position: 'top-center' });
+    }
+  }, [isError]);
+
   return (
-    <BackgroundImage
-      backgroundUrl="https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_"
-      data-test-id={surveyScreenTestIds.backgroundImage}
-    >
-      <div className="flex flex-col">
-        <button className="mt-[37px] ml-[39px] p-1" data-test-id={surveyScreenTestIds.backButton}>
-          <ArrowBack />
-        </button>
-        <div className="w-1/2 self-center pt-36">
-          <img
-            src="https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_l"
-            className="w-full h-[302px] rounded-[12px] object-cover"
-            alt="survey"
-            data-test-id={surveyScreenTestIds.coverImage}
-          />
-          <p className="text-white text-x-large font-extrabold pt-8" data-test-id={surveyScreenTestIds.title}>
-            Working from home Check-In
-          </p>
-          <p
-            className="text-white text-regular tracking-survey-tight opacity-60 pt-2"
-            data-test-id={surveyScreenTestIds.description}
-          >
-            We would like to know how you feel about our work from home (WFH) experience.
-          </p>
-          <div className="pt-8">
-            <ElevatedButton isFullWidth type="submit" data-test-id={surveyScreenTestIds.startSurveyButton}>
-              Start Survey
-            </ElevatedButton>
+    <div>
+      <ToastContainer />
+      <BackgroundImage
+        backgroundUrl={isLoading ? undefined : survey?.coverImageUrl}
+        data-test-id={surveyScreenTestIds.backgroundImage}
+      >
+        {isLoading || isError ? (
+          <div></div>
```

Comment:
Could we just return it, instead of using `ternary`?

 ```suggestion
        if (isLoading || isError) {
          return;
        };
```

Reply:
@tyrro sorry, I cannot use `if` in render method. Instead, I try `ternary` by following [this answer](https://stackoverflow.com/a/40477289) 🙏